src/main/java/org/cryptomator/cloudaccess/webdav/WebDavClient.java

```diff
@@ -399,9 +415,18 @@ void checkServerCompatibility() throws ServerNotWebdavCompatibleException {
 		}
 	}
 
-	void tryAuthenticatedRequest() throws UnauthorizedException {
+	PropfindEntryItemData tryAuthenticatedRequest() throws UnauthorizedException {
 		LOG.trace("tryAuthenticatedRequest");
-		itemMetadata(CloudPath.of("/"));
+		return loadPropfindItem(CloudPath.of("/"));
+	}
+
+	void canUseCaching(PropfindEntryItemData propfindEntryItemData) throws UnauthorizedException {
```

Comment:
changing this object's state from within a method called "canSomthing" feels like a side effect... Maybe we can find out capabilities from within a WebDavClientBuilder and whether etag caching is used or not be an immutable property set in the WebDavClients constructor?